File: app/src/main/java/com/dmi/perfectreader/app/AppThreads.java
package com.dmi.perfectreader.app;

import com.dmi.util.MainThreads;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.Future;

public class AppThreads {
    private static ExecutorService ioExecutor = Executors.newSingleThreadExecutor();

    public static Future<?> postIOTask(Runnable task) {
        return ioExecutor.submit(task);
    }

    public static void postUITask(Runnable task) {
        MainThreads.post(task);
    }
}

File: app/src/main/kotlin/com/dmi/perfectreader/app/AppThreads.kt
package com.dmi.perfectreader.app

import com.dmi.util.concurrent.Threads
import java.util.concurrent.Executors
import java.util.concurrent.Future

object AppThreads {
    private val ioExecutor = Executors.newSingleThreadExecutor()

    fun postUITask(task: () -> Unit): Future<*> {
        return Threads.postUITask(task)
    }

    fun postIOTask(task: () -> Unit): Future<*> {
        return ioExecutor.submit(task)
    }
}
